scripts/copy_from_pqclean/src/kem/family/kem_scheme.c

```diff
@@ -31,15 +31,52 @@ OQS_KEM *OQS_KEM_{{ family }}_{{ scheme['scheme'] }}_new() {
 	return kem;
 }
 
-extern int PQCLEAN_{{ scheme['pqclean_scheme_c']|upper }}_{{ scheme['default_implementation']|upper }}_crypto_kem_keypair(unsigned char *pk, unsigned char *sk);
-extern int PQCLEAN_{{ scheme['pqclean_scheme_c']|upper }}_{{ scheme['default_implementation']|upper }}_crypto_kem_enc(unsigned char *ct, unsigned char *ss, const unsigned char *pk);
-extern int PQCLEAN_{{ scheme['pqclean_scheme_c']|upper }}_{{ scheme['default_implementation']|upper }}_crypto_kem_dec(unsigned char *ss, const unsigned char *ct, const unsigned char *sk);
+    {%- for impl in scheme['metadata']['implementations'] if impl['name'] == scheme['default_implementation'] %}
+
+        {%- if impl['signature_keypair'] %}
+           {%- set cleankeypair = scheme['metadata'].update({'default_keypair_signature': impl['signature_keypair']}) -%}
```

Comment:
I'm a bit confused about what's going on here. Are the set variables cleankeypair, cleanenc, cleandec ever used? If so I missed it. And you're updating the scheme['metadata'] data structure here in the template? Shouldn't that logic be back in the Python file that created that data structure in the first place?

Reply:
The variables indeed are never used: The purpose of the `set` statement is only to trigger the side-effect of setting the scheme-array-based 'variables' ("default_keypair_signature" in this example) as those 'variables' are needed in later template fields (beyond the reach of jinja's normal, [limited scoping rules](https://jinja.palletsprojects.com/en/2.11.x/templates/)). I also left this code intentionally in this place to ease readability as to what the logic flow is (based on presence/absence of specific `impl` fields). Looks like I failed :-(